x/staging/posts/keeper/keeper.go

```diff
@@ -65,19 +61,9 @@ func (k Keeper) SavePost(ctx sdk.Context, post types.Post) {
 	// Save the post
 	store.Set(types.PostStoreKey(post.PostID), k.cdc.MustMarshalBinaryBare(&post))
 
-	// Check if the postID got an associated post, if not, increment the number of posts
-	if !store.Has(types.PostIndexedIDStoreKey(post.PostID)) {
-		// Retrieve the total number of posts, if null it will be equal to 0
-		postIndex := types.PostIndex{Value: 0}
-		if store.Has(types.PostTotalNumberPrefix) {
-			k.cdc.MustUnmarshalBinaryBare(store.Get(types.PostTotalNumberPrefix), &postIndex)
-		}
-
-		postIndex = types.PostIndex{Value: postIndex.Value + 1}
-
-		// Save the new incremental ID of the post and update the total number of posts
-		store.Set(types.PostIndexedIDStoreKey(post.PostID), k.cdc.MustMarshalBinaryBare(&postIndex))
-		store.Set(types.PostTotalNumberPrefix, k.cdc.MustMarshalBinaryBare(&postIndex))
+	// Save the query key if the key does not exist
+	if !store.Has(types.PostQueryStoreKey(post.Subspace, post.PostID)) {
+		store.Set(types.PostQueryStoreKey(post.Subspace, post.PostID), []byte(post.PostID))
 	}
```

Comment:
This can and should be simplified to avoid calling a method twice: 
```suggestion
	subspaceKey := types.PostQueryStoreKey(post.Subspace, post.PostID)
	if !store.Has(subspaceKey) {
		store.Set(subspaceKey, []byte(post.PostID))
	}
```